fabric-resource-conditions-api-v1/src/main/java/net/fabricmc/fabric/api/resource/conditions/v1/ResourceConditions.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.resource.conditions.v1;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import com.google.gson.JsonParseException;
+
+import net.fabricmc.fabric.impl.resource.conditions.ResourceConditionsImpl;
+
+import net.minecraft.util.Identifier;
+
+import net.minecraft.util.JsonHelper;
+
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Registration and access to JSON resource conditions.
+ * A JSON resource condition is a named {@code Predicate<JsonElement>} that can get tested when loading JSON resources.
+ *
+ * <p>TODO: fabric defines the following resource conditions: ...
+ *
+ * <p>TODO: fabric supports resource conditions in the following places: ...
+ */
+public final class ResourceConditions {
+	private static final Map<Identifier, Predicate<JsonObject>> REGISTERED_CONDITIONS = new ConcurrentHashMap<>();
+	/**
+	 * The key fabric uses to identify resource conditions in a JSON object.
+	 */
+	public static final String CONDITIONS_KEY = "fabric:conditions";
+	public static final String CONDITION_ID = "condition";
+
+	/**
+	 * Register a new resource condition.
+	 *
+	 * @throws IllegalArgumentException If a resource condition is already registered with the same name.
+	 */
+	public static void register(Identifier identifier, Predicate<JsonObject> condition) {
+		Objects.requireNonNull(identifier, "Identifier may not be null.");
+		Objects.requireNonNull(condition, "Condition may not be null.");
+
+		if (REGISTERED_CONDITIONS.put(identifier, condition) != null) {
+			throw new IllegalArgumentException("Duplicate JSON condition registration with id " + identifier);
+		}
+	}
+
+	/**
+	 * Get the resource condition with the passed name, or {@code null} if none is registered (yet).
+	 */
+	@Nullable
+	public static Predicate<JsonObject> get(Identifier identifier) {
+		return REGISTERED_CONDITIONS.get(identifier);
+	}
+
+	/**
+	 * Check if the passed JSON object either has no {@code fabric:conditions} tag, or all of its conditions match.
+	 * This should be called for objects that may contain a conditions entry.
+	 */
+	public static boolean objectMatchesConditions(JsonObject object) {
```

Comment:
Maybe this? I am not so sure about this...

```suggestion
	public static boolean resourceMatchesConditions(JsonObject object) {
```

Reply:
The idea is that these conditions can also be used for nested objects, not necessarily the resource itself. For example if you only want some outputs to be produced if a mod is loaded.